src/routing_core.rs

```diff
@@ -743,9 +743,313 @@ impl RoutingCore {
+                                                        opt_connection = Some(connection.clone());
+                                                        break;
+                                                    },
+                                                    None => {},
+                                                }
+                                            }
+                                        },
+                                    }
+                                }
+                            },
+                            ExpectedConnection::Response(ref response, _) => {
+                                for (key, value) in self.expected_connections.iter() {
+                                    match key {
+                                        &ExpectedConnection::Request(ref request) => {
+                                            // We are the network-side with a ConnectRequest, Node B
+                                            // on diagram of RFC-0011.
+                                            if response.receiver_fob.name() == self.id().name() &&
+                                                    hello.public_id == request.requester_fob {
+                                                match value.0 {
+                                                    Some(connection) => {
+                                                        opt_connection =
+                                                            Some(unknown_connection.clone());
+                                                        break;
+                                                    },
+                                                    None => {},
+                                                }
+                                            }
+                                        },
+                                        &ExpectedConnection::Response(_, _) => {
+                                            // Don't match on ConnectResponse.
+                                        },
+                                    }
+                                }
+                            },
+                        }
+                    },
+                    None => {},
+                };
+
+                match opt_connection {
+                    Some(connection) => {
+                        if self.add_peer(ConnectionName::Routing(
+                                hello.public_id.name()), connection, hello.public_id.clone()) {
+                            if connection != unknown_connection {
+                                self.action_sender.send(::action::Action::SendConfirmationHello(
+                                    connection, ::types::Address::Node(hello.public_id.name())));
+                                let _ = self.action_sender.send(::action::Action::DropConnections(
+                                    vec![unknown_connection]));
+                            } else {
```

Comment:
the "confirmation of the hello" is not explicitly checked; which raises some questions, as if we already have a hello stored in unknown connections, then we drop any subsequent action on a hello.  So we need to check within handle_hello whether it is a consistent confirmation and do match.